src/parquet/reader-test.cc

```diff
@@ -84,36 +84,56 @@ TEST_F(TestAllTypesPlain, TestBatchRead) {
   ASSERT_FALSE(col->HasNext());
 }
 
+void verifyScanner(Int32Scanner& scanner) {
+  int32_t val;
+  bool is_null;
+  for (size_t i = 0; i < 8; ++i) {
+    ASSERT_TRUE(scanner.HasNext());
+    ASSERT_TRUE(scanner.NextValue(&val, &is_null));
+    ASSERT_FALSE(is_null);
+  }
+  ASSERT_FALSE(scanner.HasNext());
+  ASSERT_FALSE(scanner.NextValue(&val, &is_null));
+}
```

Comment:
My specific point is that we really need a unit test that reproduces this bug, and so the moment we nix these temporary tests we have to go back and retroactively write the unit tests we probably should have written the first time around. I only wrote these tests here to help me do the refactoring in PARQUET-433 (which would have severely blocked progress if not gotten through quickly), but not as a long term solution.